UI/ChatGPT/src/ChatGPT/Services/ChatService.cs

```diff
@@ -1,39 +1,60 @@
-﻿using ChatGPT.Business;
-using OpenAI.Interfaces;
-using OpenAI.ObjectModels;
-using OpenAI.ObjectModels.RequestModels;
-using OpenAI.ObjectModels.ResponseModels;
+﻿using System.ClientModel;
 using System.Runtime.CompilerServices;
 using System.Text;
 
 namespace ChatGPT.Services;
-public class ChatService(IChatCompletionService client) : IChatService
+public class ChatService : IChatService
 {
-	private readonly IChatCompletionService _client = client;
-
 	private const string systemPrompt = "You are Uno ChatGPT Sample, a helpful assistant helping users to learn more about how to develop using Uno Platform.";
 
+	private readonly ChatClient _client;
+
+	public ChatService(IOptions<AppConfig> appConfig)
+	{
+		if(appConfig.Value.ApiKey is null)
+		{
+			throw new InvalidOperationException("You must define an OpenAI API key in appsettings.json file.");
+		}
+
+		_client = new ChatClient("gpt-3.5-turbo", appConfig.Value.ApiKey);
+	}
+
 	public async ValueTask<ChatResponse> AskAsync(ChatRequest chatRequest, CancellationToken ct = default)
 	{
 		try
 		{
 			var request = ToCompletionRequest(chatRequest);
-			var result = await _client.CreateCompletion(request, cancellationToken: ct);
-
-			if (result.Successful)
+			ChatCompletion result = await _client.CompleteChatAsync(request);
+			ChatResponse? chatResponse = null;
+			switch (result.FinishReason)
```

Comment:
```suggestion
			var result = await _client.CompleteChatAsync(request);
			return result.FinishReason switch
```

Reply:
Keeping that line as `ChatCompletion result = await _client.CompleteChatAsync(request);` otherwise the type returned is 
![image](https://github.com/unoplatform/Uno.Samples/assets/114886335/edf5b04c-66f7-4377-a4e3-2eadb11a6183)